### ISSUE TITLE
Latest linked saltstack has error

### ISSUE THREAD

**Issue by eseglem** [plus3it/watchmaker#1973]
Sorry, this might not be the best repository to post this issue on, but our issue most likely relates to this PR that was merged yesterday: https://github.com/plus3it/watchmaker/pull/1903 It appears to point at a new saltstack where the underlying error is. I just wasn't sure which repo includes the appropriate saltstacks to submit an issue against. 

We use the following command, approximately, for nightly deploys in our dev environment: 
```
watchmaker -c https://raw.githubusercontent.com/plus3it/watchmaker/master/src/watchmaker/static/config.yaml -e test -t "$HOSTNAME" --no-reboot
```
 (We reboot later on)

This runs for a few minute and then ends up failing at:
```
Salt return (AttributeError): {'jid': '20220302221827869279', 'return': ["Cannot extend ID 'install oscap packages' in 'base:scap.oscap.scan'. It is not part of the high state.\nThis is likely due to a missing include statement or an incorrectly typed ID.\nEnsure that a state with an ID of 'install oscap packages' is available\nin environment 'base' and to SLS 'scap.oscap.scan'"], 'retcode': 1, 'out': 'highstate', 'id': 'hostname', 'fun': 'state.highstate', 'fun_args': [], 'success': True}
Setting selinux back to enforcing mode
Command: setenforce enforcing
Command retcode: 0
Execution of the workers cadence has failed.

Traceback (most recent call last):
  File "/usr/local/bin/watchmaker", line 10, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/watchmaker/cli.py", line 110, in main
    sys.exit(watchmaker_client.install())
  File "/usr/local/lib/python3.6/site-packages/watchmaker/__init__.py", line 500, in install
    workers_manager.worker_cadence()
  File "/usr/local/lib/python3.6/site-packages/watchmaker/managers/worker_manager.py", line 59, in worker_cadence
    worker.install()
  File "/usr/local/lib/python3.6/site-packages/watchmaker/workers/salt.py", line 841, in install
    self.process_states(self.salt_states, self.exclude_states)
  File "/usr/local/lib/python3.6/site-packages/watchmaker/workers/salt.py", line 672, in process_states
    indent=4
watchmaker.exceptions.WatchmakerError: Salt state execution failed:
- 'Cannot extend ID ''install oscap packages'' in ''base:scap.oscap.scan''. It is
    not part of the high state.

    This is likely due to a missing include statement or an incorrectly typed ID.

    Ensure that a state with an ID of ''install oscap packages'' is available

    in environment ''base'' and to SLS ''scap.oscap.scan'''
```

**Comment by lorengordon** [comment]
Hi @eseglem! Thanks for reporting. That problem has been resolved on the "main" branch. The patch has not yet been included in a watchmaker release though. How are you installing watchmaker?

Since you are simply pointing at the default config, I might suggest removing that argument. Watchmaker bundles a default config that is tested to work with the released version... E.g.

```
watchmaker -e test -t "$HOSTNAME" --no-reboot
```

Or alternatively, you could point to a pinned version of the released config:

```
watchmaker -c https://raw.githubusercontent.com/plus3it/watchmaker/0.23.4/src/watchmaker/static/config.yaml -e test -t "$HOSTNAME" --no-reboot 
```

**Comment by eseglem** [comment]
Oh, yeah, that would probably do it. We just `pip install watchmaker` and the mismatch with release and the yaml in main could certainly trip things up. 

We have all the arguments set up as env variables, since we don't always point to that url. It just defaults to pointing at that url if we don't point at something else. Probably could adjust it to only pass in `-c` if the var is set though. Since a default to a pinned version wouldn't make much sense.

**Comment by lorengordon** [comment]
Yep, mismatch then between the released/stable watchmaker version and the unreleased config.yaml. We try to be pretty careful about testing upgrades to the salt version in the default config, as they often expose some kind of issue in the salt formulas. Which is how I know we already accounted for that exact problem in the main branch. Feel free to close the issue if addressing the mismatch unblocks you.

If you really wanted, you could `pip install git+https://github.com/plus3it/watchmaker.git` to test using the unreleased main branch.

**Comment by lorengordon** [comment]
We actually do have that already:
https://github.com/plus3it/watchmaker/blob/26cb43e5e52d1f9b164e6e365dc21a251307017e/src/watchmaker/static/config.yaml#L1

It's just also dependent on a released version, so it's not effective at these release/unrelease mismatches... We'll increment it for the next release.
